apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/faucet/components/SpinWheel.tsx

```diff
@@ -0,0 +1,191 @@
+"use client";
+
+import { CopyTextButton } from "@/components/ui/CopyTextButton";
+import { Spinner } from "@/components/ui/Spinner/Spinner";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { THIRDWEB_ENGINE_FAUCET_WALLET } from "@/constants/env";
+import { CustomConnectWallet } from "@3rdweb-sdk/react/components/connect-wallet";
+import { useMutation } from "@tanstack/react-query";
+import type { ChainMetadataWithServices } from "app/(dashboard)/(chain)/types/chain";
+import { thirdwebClient } from "lib/thirdweb-client";
+import { useRouter } from "next/navigation";
+import Confetti from "react-confetti";
+import { toast } from "sonner";
+import { type ISpinWheelProps, SpinWheel } from "spin-wheel-game";
+import { defineChain } from "thirdweb";
+import { useActiveAccount, useWalletBalance } from "thirdweb/react";
+
+const segments = [
+  { segmentText: "Come back tomorrow", segColor: "#f771c9" }, // Mid-dark, optimized for dark mode
+  { segmentText: "0.1", segColor: "#f65ac0" }, // Lightest, optimized for dark mode
+  { segmentText: "Come back tomorrow", segColor: "#f542b7" }, // Slightly darker, optimized
+  { segmentText: "0.01", segColor: "#f32bae" }, // Mid-light, optimized for dark mode
+  { segmentText: "0.05", segColor: "#f213a5" }, // Mid, optimized for dark mode
+  { segmentText: "0.01", segColor: "#da1195" }, // Mid-dark, optimized for dark mode
+  { segmentText: "0.05", segColor: "#c20f84" }, // Darker, optimized for dark mode
+  { segmentText: "0.01", segColor: "#a90d73" }, // Even darker, optimized
+  { segmentText: "0.05", segColor: "#910b63" }, // Near darkest, optimized for dark mode
+  { segmentText: "0.01", segColor: "#790a53" }, // Lightest, optimized for dark mode
+  { segmentText: "0.05", segColor: "#610842" }, // Slightly darker, optimized
+  { segmentText: "0.01", segColor: "#490632" }, // Mid-light, optimized for dark mode
+];
+
+function formatTime(seconds: number) {
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor((seconds % 3600) / 60);
+
+  if (hours === 0 && minutes <= 1) {
+    return "1 minute";
+  }
+
+  if (seconds > 60 && seconds < 3600) {
+    return `${minutes}m`;
+  }
+
+  return `${hours}h ${minutes}m`;
+}
```

Comment:
could you use: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat here?

Reply:
done